main.js

```diff
@@ -69,4 +158,87 @@ define(function (require, exports, module) {
         // Register nohost server with highest priority
         LiveDevServerManager.registerServer({ create: _getServer }, 9001);
     });
+
+    AppInit.appReady(function (){
+        init();
+        // Once the app has loaded our file,
+        // and we can be confident the editor is open,
+        // get a reference to it and attach our "onchange"
+        // listener to codemirror
+        codeMirror = EditorManager.getActiveEditor()._codeMirror;
+
+        parentWindow.postMessage(JSON.stringify({
+            type: "bramble:change",
+            sourceCode: codeMirror.getValue()
+        }), "*");
+
+        codeMirror.on("change", function(e){
+            parentWindow.postMessage(JSON.stringify({
+                type: "bramble:change",
+                sourceCode: codeMirror.getValue()
+            }), "*");
+        });
+    });
+
+    // Eventually, we'll listen for a message from
+    // thimble containing the make's initial code.
+    // For now, we are defaulting to thimble's starter
+    // make.
+    exports.initExtension = function() {
+        var deferred = new $.Deferred();
+        var data;
+
+        function _getInitialDocument(e) {
+            try {
+                data = e.data || null;
+                data = JSON.parse(data);
+                data = data || {};
+            } catch(err) {
+                // Quick fix: Ignore the 'process-tick' message being sent
+                if(e.data === 'process-tick') {
+                    return;
+                }
+
+                console.error("Parsing message from thimble failed: ", err);
+
+                deferred.reject();
+                return;
+            }
+
+            // Remove the listener after we confirm the event is the
+            // one we're waiting for
+            if (data.type !== "bramble:init") {
+                return;
+            }
+            window.removeEventListener("message", _getInitialDocument);
+
+            fs.writeFile(
+                '/index.html',
+                data.source ? data.source : defaultHTML,
+                function(err) {
+                    if (err) {
+                        deferred.reject();
+                        return;
+                    }
+
+                    deferred.resolve();
+                }
+            );
+        }
+        window.addEventListener("message", _getInitialDocument);
+
+        // Signal to thimble that we're waiting for the
+        // initial make source code
+        window.parent.postMessage(JSON.stringify({
+            type: "bramble:init"
+        }), "*");
+
+        return deferred.promise();
+    };
+
+    // Define public API
```

Comment:
These exports can be removed, but should be included in whatever file you put the hide logic in.